builtin/logical/pki/cert_util_test.go

```diff
@@ -158,3 +158,34 @@ func TestPki_MultipleOUs(t *testing.T) {
 		t.Fatalf("Expected %v, got %v", expected, actual)
 	}
 }
+
+func TestPki_PermitFQDNs(t *testing.T) {
+	var b backend
+	fields := addCACommonFields(map[string]*framework.FieldSchema{})
+
+	apiData := &framework.FieldData{
+		Schema: fields,
+		Raw: map[string]interface{}{
+			"common_name": "example.com.",
+			"ttl":         3600,
+		},
+	}
+	input := &inputBundle{
+		apiData: apiData,
+		role: &roleEntry{
+			AllowAnyName: true,
```

Comment:
I think you need EnforceHostnames=true for the regex to be used.

Reply:
https://github.com/hashicorp/vault/blob/2983ae924b966dbf574895391a6722ea0b5f342e/builtin/logical/pki/cert_util.go#L808

Did also fail in before I changed the regex, but more checks are better.
I'll add `EnforeHostnames` as well.